File: 251126/잔해물을 덮기 위한 사각형의 최소 넓이/minimum-area-of-rectangle-to-cover-debris.py
x1, y1, x2, y2 = [0] * 2, [0] * 2, [0] * 2, [0] * 2
x1[0], y1[0], x2[0], y2[0] = map(int, input().split())
x1[1], y1[1], x2[1], y2[1] = map(int, input().split())

OFFSET = 1000
arr = [[0] * (OFFSET * 2 + 1) for _ in range(OFFSET * 2 + 1)]

for i in range(x1[0], x2[0]):
    for j in range(y1[0], y2[0]):
        arr[i + OFFSET][j + OFFSET] = 1

for i in range(x1[1], x2[1]):
    for j in range(y1[1], y2[1]):
        arr[i + OFFSET][j + OFFSET] = 2

min_x = OFFSET * 2 + 1
max_x = -1
min_y = OFFSET * 2 + 1
max_y = -1

for x in range(len(arr)):
    for y in range(len(arr)):
        if arr[x][y] == 1:
            if min_x > x:
                min_x = x
            if max_x < x:
                max_x = x
            if min_y > y:
                min_y = y
            if max_y < y:
                max_y = y

count = 0
for x in range(min_x, max_x + 1):
    for y in range(min_y, max_y + 1):
        count += 1

print(count)

File: 260319/잔해물을 덮기 위한 사각형의 최소 넓이/minimum-area-of-rectangle-to-cover-debris.js
const OFFSET = 1000;
const MAX_SIZE = OFFSET * 2 + 1;

const fs = require("fs");
const input = fs.readFileSync(0).toString().trim().split('\n');

const rect1 = input[0].split(' ').map(Number);
const rect2 = input[1].split(' ').map(Number);

let grid = Array.from({ length: MAX_SIZE }, () => Array(MAX_SIZE).fill(0));

let [rect1X1, rect1Y1, rect1X2, rect1Y2] = rect1;
let [rect2X1, rect2Y1, rect2X2, rect2Y2] = rect2;

rect1X1 = Number(rect1X1) + OFFSET;
rect1X2 = Number(rect1X2) + OFFSET;
rect1Y1 = Number(rect1Y1) + OFFSET;
rect1Y2 = Number(rect1Y2) + OFFSET;

rect2X1 = Number(rect2X1) + OFFSET;
rect2X2 = Number(rect2X2) + OFFSET;
rect2Y1 = Number(rect2Y1) + OFFSET;
rect2Y2 = Number(rect2Y2) + OFFSET;

for (let x = rect1X1; x < rect1X2; x++) {
    for (let y = rect1Y1; y < rect1Y2; y++) {
        grid[x][y] = 1;
    }
}

for (let x = rect2X1; x < rect2X2; x++) {
    for (let y = rect2Y1; y < rect2Y2; y++) {
        grid[x][y] = 2;
    }
}

let minX = MAX_SIZE, minY = MAX_SIZE;
let maxX = 0, maxY = 0;
let isExist = false;
for (let i = 0; i < MAX_SIZE; i++) { // 행
    for (let j = 0; j < MAX_SIZE; j++) { // 열
        if (grid[i][j] === 1) {
            isExist = true;
            minX = Math.min(minX, i);
            minY = Math.min(minY, j);
            maxX = Math.max(maxX, i);
            maxY = Math.max(maxY, j);
        }
    }
}


let answer = 0;

if (!isExist) {
    console.log(0);
} else {
    answer = (maxX - minX + 1) * (maxY - minY + 1);
    console.log(answer);
}